src/diff/children.js

```diff
@@ -81,15 +81,16 @@ export function diffChildren(
 					// Either oldVNode === undefined or oldChildrenLength > 0,
 					// so after this loop oldVNode == null or oldVNode is a valid value.
 					for (j = 0; j < oldChildrenLength; j++) {
-						oldVNode = oldChildren[j];
+						const idx = i === 0 ? j : (j + i - 1) % oldChildrenLength;
```

Comment:
We could transform this to

```
const idx = j + i === 0 ? j : (j + i - 1) % oldChildrenLength;
```

But this seems to goof up some focus tests:
- should maintain focus when removing elements around input
- Fragment - should preserve state of children nested at same level (adds stateful upate)
- Fragment - should preserve state with reordering in multiple levels (adds stateful upate)
- keys - should swap children efficiently (makes it more efficient, one operation instead of 3)
- should move keyed children to the end of the list  (makes it more efficient, one operation instead of 3)
- should reverse children effectively more efficient (11 --> 9 operations)

Reply:
The last 3 items are spot on! 👍 Wondering about the other 3 what's up with those?